### ISSUE TITLE
Comma Dangle error if prettier ignore is set to ignore last property in object

### ISSUE THREAD

**Issue by govindrai** [prettier/prettier#5678]
**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAnOAzXNACwBUIBrBABQEMBPAGwhoBNMBeTYAHSk38ww6ABzhJMAclwE4xMpSgSANLwGY0AVwBG4mEQCWaAHQB9fSxV8B2CAzGCDxm3cv9uMAPQfMw3DBj6cNgAtPoA5lAQuKoCcAAewuKRAO6YANSYABQAbAAMmABUmHmFmABMACylAJy5AJRKmF5ZtZgs9Gh1MQC+ANwgSiAQwgHoyKA02DbJtLhQaMggNABuEOYDIFrYNGCUMADKwjv6UGHIMNgacINEMAC2DADqBvBoR2Bw+who+gHLv3RFmA0AtBic0EEYFRtmE7jRkHgaAwIYMAFZoOIAIW2uzgBxodzgABkTnAEUiUSB0XF9icwnYAIoaCDwcnI64gI7YCHYRZaGhaRjQDa+E4wR7mPTIAAcuUGvggEMe22Ei18siCyzJg1wAEcNPpcNCaLD4UhEezBhC7vpzpcOT9TozmazzRSOTABRKWFKkGVBhcaPoGHSAMIQO5wxaRKDakAaCEkAULN3s7rdIA)
```sh
--parser babylon
```

**Input:**
```jsx
const refreshTokenPayload = {
    type: 'refreshToken',
    sub: this._id,
    role: this.role,
  	// prettier-ignore
    exp: now + (60 * 60 * 24 * 90), // (90 days)
  };
```

**Output:**
```jsx
const refreshTokenPayload = {
  type: "refreshToken",
  sub: this._id,
  role: this.role,
  // prettier-ignore
  exp: now + (60 * 60 * 24 * 90)
};

```

**Expected behavior:**
```jsx
const refreshTokenPayload = {
  type: "refreshToken",
  sub: this._id,
  role: this.role,
  // prettier-ignore
  exp: now + (60 * 60 * 24 * 90), // <-- prettier removes the comma and then the linter blows up
};

```

FYI, I used prettier ignore so I could keep my parentheses https://github.com/prettier/prettier/issues/187

**Comment by j-f1** [comment]
Also fails with trailing commas enabled:

**Prettier 1.15.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAnOAzXNACwBUIBrBABQEMBPAGwhoBNMBeTYAHSk38ww6ABzhJMAclwE4xMpSgSANLwGY0AVwBG4mEQCWaAHQB9fSxV8B2CAzGCDxm3cv9uMAPQfMw3DBj6cNgAtPoA5lAQuKoCcAAewuKRAO6YANSYABQAbAAMmABUmHmFmABMACylAJy5AJRKmF5ZtZgs9Gh1MQC+ANwgSiAQwgHoyKA02DbJtLhQaMggNABuEOYDIFrYNGCUMADKwjv6UGHIMNgacINEMAC2DADqBvBoR2Bw+who+gHLv3RFmA0AtBic0EEYFRtmE7jRkHgaAwIYMAFZoOIAIW2uzgBxodzgABkTnAEUiUSB0XF9icwnYAIoaCDwcnI64gI7YCHYRZaGhaRjQDa+E4wR7mPTIAAcuUGvggEMe22Ei18siCyzJg1wAEcNPpcNCaLD4UhEezBhC7vpzpcOT9TozmazzRSOTABRKWFKkGVBhcaPoGHSAMIQO5wxayACsGw0EJIAoWbvZ3W6QA)
```sh
--parser babylon
--trailing-comma es5
```

**Input:**
```jsx
const refreshTokenPayload = {
    type: 'refreshToken',
    sub: this._id,
    role: this.role,
  	// prettier-ignore
    exp: now + (60 * 60 * 24 * 90), // (90 days)
  };
```

**Output:**
```jsx
const refreshTokenPayload = {
  type: "refreshToken",
  sub: this._id,
  role: this.role,
  // prettier-ignore
  exp: now + (60 * 60 * 24 * 90) // (90 days)
};

```

**Comment by rcollette** [comment]
Maybe related, but when an javascript/typescipt array is formatted, the closing bracket is placed on another line, by itself, and the linter then wants a comma after the last entry.   The linter is actually expecting the closing bracket to be on the same line as the last element.

Personally, I think this is an issue with the linter.   I like the closing bracket on its own line and prefer not to have a comma since there is nothing to separate/delineate after the last element.   But that probably gets into a battle of personal preferences.

**Comment by lydell** [comment]
@rcollette That seems unrelated to this issue. Regarding linters, we recommend turning all conflicting rules off. See https://prettier.io/docs/en/eslint.html.

**Comment by Pointotech** [comment]
I'm seeing this bug too, in a file with this content:

    export default {
      // prettier-ignore
      protagonist: "  0\r\n" +
                   "0 00\r\n" +
                   "00000\r\n" +
                   "0 0\r\n" +
                   "0 0",

      // prettier-ignore
      wall: "00000\r\n" +
            "00000\r\n" +
            "00000\r\n" +
            "00000\r\n" +
            "00000",

      // prettier-ignore
      cheese: "0\r\n" +
              " 0\r\n" +
              "000\r\n" +
              "00 0\r\n" +
              "00000",

      // prettier-ignore
      enemy: "0   0\r\n" +
             "00 00\r\n" +
             "00000\r\n" +
             "0 0 0\r\n" +
             "00000",

      // prettier-ignore
      home: "00000\r\n" +
            "0   0\r\n" +
            "0   0\r\n" +
            "0   0\r\n" +
            "00000",

      // prettier-ignore
      dog: "00 00\r\n" +
           "00000\r\n" +
           "0   0\r\n" +
           "0 0 0\r\n" +
           " 000 "
    }

I have `trailingComma: true` in my `.prettierrc` file, and it works correctly for every other file in the project. In fact, it even works in this file, as long as I do one of two things that I don't want to do and shouldn't need to do.

1) Add another, non-`prettier-ignore` property, before the end of the item, like this:

        export default {
          // ...

          // prettier-ignore
          dog: "00 00\r\n" +
               "00000\r\n" +
               "0   0\r\n" +
               "0 0 0\r\n" +
               " 000 ",

          foo: "bar",
        }

In this case, since the last item does not have `prettier-ignore` on it, the bug is avoided. This is not acceptable because it should not be necessary to pollute the code with dummy variables in order to get normalized formatting.

2) Remove `prettier-ignore` from the last item, like this:

        export default {
          // ...

          dog: "00 00\r\n" + "00000\r\n" + "0   0\r\n" + "0 0 0\r\n" + " 000 ",
        }

In this case, since the last item does not have `prettier-ignore` on it, the bug is avoided. This is not acceptable because it destroys the intentionally weird formatting that `prettier-ignore` should preserve.

**Comment by mesqueeb** [comment]
I am facing the same problem......

**Comment by mesqueeb** [comment]
@evilebottnawi why was this closed?

**Comment by fisker** [comment]
Fixed by #8111
